nemo_aligner/data/nlp/datasets.py

```diff
@@ -26,6 +26,59 @@
 from nemo.utils import logging
 
 
+class KnowledgeDistillationDataset(Dataset):
+    """The knowledge distillation dataset takes in raw tokens, labels, loss masks, and the teacher models' predictive top tokens & logits.
+    """
+
+    def __init__(
+        self, cfg, tokenizer, name, data_prefix, documents, data, seq_length, seed, drop_last=True,
+    ):
+        super().__init__()
+        self.cfg = cfg
+        self.name = name
+        self.data = data
+        self.seq_length = seq_length
+
+        self.nograd_length = 32
```

Comment:
leaving a note here to double check this does what we think since I'm suspicious it might not do what we want

Reply:
I asked Shengyang about this a while back and he explained that this is the dummy length we use for examples whose actual sequence length is larger than max_length. We do the same for DPO; see https://github.com/NVIDIA/NeMo-Aligner/blob/fb49c277af6e59455e9449c05b7aeacd1fcab862/nemo_aligner/data/nlp/datasets.py#L273